pkg/controllers/job/job_controller_actions.go

```diff
@@ -296,13 +296,17 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 	}
 
 	var jobCondition batch.JobCondition
+	oldStatus := job.Status
 	if !syncTask {
 		if updateStatus != nil {
-			if updateStatus(&job.Status) {
-				job.Status.State.LastTransitionTime = metav1.Now()
-				jobCondition = newCondition(job.Status.State.Phase, &job.Status.State.LastTransitionTime)
-				job.Status.Conditions = append(job.Status.Conditions, jobCondition)
-			}
+
+			job.Status.State.LastTransitionTime = metav1.Now()
+			jobCondition = newCondition(job.Status.State.Phase, &job.Status.State.LastTransitionTime)
+			job.Status.Conditions = append(job.Status.Conditions, jobCondition)
+		}
+		if reflect.DeepEqual(job.Status, oldStatus) {
```

Comment:
should we use `quality.Semantic.DeepEqual` when compare two objects.
refer to: https://github.com/volcano-sh/volcano/issues/3488

Reply:
@googs1025 : changes are done. pls check